File: backend/src/main/java/es/codeurjc/webapp14/dto/UserDTO.java
package es.codeurjc.webapp14.dto;

import java.util.List;

public record UserDTO(
    Long id,
    String name,
    String surname,
    String email,
    String address,
    boolean banned,
    int reports,
    String imageUrl,
    List<String> roles
) {}

File: frontend/src/app/dtos/user.dto.ts
import { ReviewDTO } from "./review.dto";




export interface UserDTO {
    id?: number,
    name: string,
    surname: string,
    email: string,
    address: string,
    banned: boolean,
    reports: number,
    imageUrl: String,
    roles: String[],
    showDetails?: boolean,
    showReports?: boolean,
    reportedReviews?: ReviewDTO[],
}